src/cmap/conn/mod.rs

```diff
@@ -46,9 +46,15 @@ pub struct ConnectionInfo {
     pub id: u32,
 
     /// A server-generated identifier that uniquely identifies the connection. Available on server
-    /// versions 4.2+. This may be used to correlate driver connections with server logs.
+    /// versions 4.2+. This may be used to correlate driver connections with server logs. 
+    /// If the connection ID sent by the server is too large for an i32, this will be a truncated value.
     pub server_id: Option<i32>,
 
+    /// A server-generated identifier that uniquely identifies the connection. Available on server
+    /// versions 4.2+. This may be used to correlate driver connections with server logs. Reference
+    /// this for the full server_id
```

Comment:
(feel free to tweak the phrasing if you'd like)
```suggestion
    /// versions 4.2+. This may be used to correlate driver connections with server logs. This value
    /// will not be truncated and should be used rather than `server_id`.
```